saml/src/main/java/com/linecorp/armeria/saml/SamlServiceProviderConfiguratorBuilder.java

```diff
@@ -0,0 +1,478 @@
+            this.ssoEndpoint = requireNonNull(ssoEndpoint, "ssoEndpoint");
+            return this;
+        }
+
+        /**
+         * Sets a single logout request endpoint of an identity provider.
+         */
+        public IdentityProviderConfigBuilder sloReqEndpoint(SamlEndpoint sloReqEndpoint) {
+            this.sloReqEndpoint = requireNonNull(sloReqEndpoint, "sloReqEndpoint");
+            return this;
+        }
+
+        /**
+         * Sets a single logout response endpoint of an identity provider.
+         */
+        public IdentityProviderConfigBuilder sloResEndpoint(SamlEndpoint sloResEndpoint) {
+            this.sloResEndpoint = requireNonNull(sloResEndpoint, "sloResEndpoint");
+            return this;
+        }
+
+        /**
+         * Sets an assertion consumer service URL of this service provider.
+         */
+        public IdentityProviderConfigBuilder acsEndpoint(SamlEndpoint acsEndpoint) {
+            this.acsEndpoint = requireNonNull(acsEndpoint, "acsUrl");
+            return this;
+        }
+
+        /**
+         * Sets a consumer which customizes an {@link AuthnRequest}.
+         */
+        public IdentityProviderConfigBuilder requestCustomizer(Consumer<AuthnRequest> requestCustomizer) {
+            this.requestCustomizer = requireNonNull(requestCustomizer, "requestCustomizer");
+            return this;
+        }
+
+        /**
+         * Sets an {@link SamlNameIdPolicy} to configure an {@link AuthnRequest}.
+         */
+        public IdentityProviderConfigBuilder nameIdPolicy(SamlNameIdPolicy nameIdPolicy) {
+            this.nameIdPolicy = requireNonNull(nameIdPolicy, "nameIdPolicy");
+            return this;
+        }
+
+        /**
+         * Sets this idp as a default.
+         */
+        public IdentityProviderConfigBuilder asDefault() {
+            isDefault = true;
+            return this;
+        }
+
+        /**
+         * Builds an {@link IdentityProviderConfig}.
+         */
+        private IdentityProviderConfig build() {
+            checkArgument(entityId != null, "IdP entity ID is missing");
+
+            // Use the entityId as a default key name.
+            signingKey = firstNonNull(signingKey, entityId);
+            encryptionKey = firstNonNull(encryptionKey, entityId);
+
+            return new IdentityProviderConfig(entityId,
+                                              resolveCredential(signingKey), resolveCredential(encryptionKey),
+                                              ssoEndpoint, sloReqEndpoint, sloResEndpoint, acsEndpoint,
+                                              requestCustomizer, nameIdPolicy);
+        }
+
+        /**
+         * Returns a {@link SamlServiceProviderConfigurator} which is the parent of this builder.
+         */
+        public SamlServiceProviderConfiguratorBuilder and() {
+            return SamlServiceProviderConfiguratorBuilder.this;
+        }
+    }
+
+    public class AssertionConsumerConfigBuilder {
```

Comment:
- The default constructor has to be package-local.
- Top-level and `Saml` prefix?